### ISSUE TITLE
ArgoCD Helm rollout results with infinite sync-loop

### ISSUE THREAD

**Issue by hijmen** [external-secrets/external-secrets#1026]
When creating ExternalSecrets application in ArgoCD using the Helm Chart on auto-sync, it results in an infinite sync loop.

I am using the following chart configuration:
```yaml
installCRDs: true
```

![external-secrets-issue](https://user-images.githubusercontent.com/1532024/164195028-29364cbb-7176-470c-976e-d7e564b215db.png)

If anyone else faces this issue, I have a temporarily workaround using IgnoreDifferences in ArgoCD:

```yaml
- group: apiextensions.k8s.io
  kind: CustomResourceDefinition
  name: secretstores.external-secrets.io
  jqPathExpressions:
    - .spec.conversion.webhook.clientConfig
- group: apiextensions.k8s.io
  kind: CustomResourceDefinition
  name: externalsecrets.external-secrets.io
  jqPathExpressions:
    - .spec.conversion.webhook.clientConfig
- group: apiextensions.k8s.io
  kind: CustomResourceDefinition
  name: clustersecretstores.external-secrets.io
  jqPathExpressions:
    - .spec.conversion.webhook.clientConfig
- group: admissionregistration.k8s.io
  kind: ValidatingWebhookConfiguration
  name: secretstore-validate
  jqPathExpressions:
    - .webhooks[].clientConfig
- group: admissionregistration.k8s.io
  kind: ValidatingWebhookConfiguration
  name: externalsecret-validate
  jqPathExpressions:
    - .webhooks[].clientConfig
```

**Comment by gusfcarvalho** [comment]
Hey @hijmen ! this is somewhat related to #997 where the same issue happens with Flux. Can you check if #1001 would fix ArgoCD behavior?

**Comment by hijmen** [comment]
Thanks @gusfcarvalho, it would be perfect timing if this solves it. I will test this in a few days!

**Comment by sjahl** [comment]
@gusfcarvalho I was able to test this on my argo cluster today: commit https://github.com/external-secrets/external-secrets/commit/c04350d0d03b5994cf368e6e7f79cbfdd86c2595 makes it so that the CRDs are no longer in an infinite sync loop 👍 

But, the deployment fails for other reasons, namely the webhook pod looks like its command is being invoked incorrectly and is throwing a "Error: unknown flag: --port" error. I can file another issue for that assuming it's not already known.

**Comment by gusfcarvalho** [comment]
Hi @sjahl ! What is happening is that you are using the most up-to-date helm charts at main but not the most up-to-date images! Basically you need to point your image to the `main` tag as stated here: https://external-secrets.io/v0.5.1/guides-using-latest-image/.

When we cut `v0.5.2` it should already work properly.

**Comment by gusfcarvalho** [comment]
Hey all ! This issue has been fixed and v0.5.2 released. Please check if it works as expected. Feel free to reopen this issue as well!
